app/jobs/email_petition_signatories.rb

```diff
@@ -0,0 +1,85 @@
+class EmailPetitionSignatories
+  class PleaseRetry < StandardError; end
+
+  def self.run_later_tonight(job_class, petition, requested_at, *extra_args)
```

Comment:
Responded in 58aa2db86096d49078b93a46293ff719e2e56652 (commit message has some reasoning in it so you might want to read that rather than review the total diff).